packages/fx-core/src/plugins/resource/function/index.ts

```diff
@@ -78,6 +79,14 @@ export class FunctionPlugin implements Plugin {
     return res;
   }
 
+  public async executeUserTask(func: Func, ctx: PluginContext): Promise<FxResult> {
+    this.setContext(ctx);
+    const res = await this.runWithErrorWrapper(ctx, FunctionEvent.executeUserTask, () =>
+      Promise.resolve(this.functionPluginImpl.executeUserTask(func, ctx))
```

Comment:
Send telemetry here. We can get the operation result by union telemetry of  `executeUserTask` and `addResource`.

Reply:
why add `Promise.resolve` here

Reply:
Removed.